nautobot_ssot_aristacv/diffsync/adapters/cloudvision.py

```diff
@@ -1,48 +1,111 @@
-                    self.job.log_debug(message=f"Device being loaded: {dev}")
-                new_device = self.device(name=dev["hostname"], device_id=dev["device_id"], device_model=dev["model"])
+                    self.job.log_debug(message=f"Device being loaded: {dev}. Port: {port_info}.")
+                new_device = self.device(
+                    name=dev["hostname"], serial=dev["device_id"], device_model=dev["model"], uuid=None
+                )
                 try:
                     self.add(new_device)
                 except ValidationError as err:
                     self.job.log_warning(message=f"Unable to load Device {dev['hostname']}. {err}")
                     continue
+                except ObjectAlreadyExists as err:
+                    self.job.log_warning(
+                        message=f"Duplicate device {dev['hostname']} {dev['device_id']} found and ignored. {err}"
+                    )
+                    continue
+
+                for port in port_info:
+                    if self.job.kwargs.get("debug"):
+                        self.job.log_debug(message=f"Port {port['interface']} being loaded for {dev['hostname']}.")
+                    port_mode = cloudvision.get_interface_mode(client=self.conn, dId=dev["device_id"], interface=port)
+                    transceiver = cloudvision.get_interface_transceiver(
+                        client=self.conn, dId=dev["device_id"], interface=port
+                    )
+                    if transceiver == "Unknown":
+                        # Breakout transceivers, ie 40G -> 4x10G, shows up as 4 interfaces and requires looking at base interface to find transceiver, ie Ethernet1 if Ethernet1/1
+                        base_port_name = re.sub(r"/\d", "", port["interface"])
+                        transceiver = cloudvision.get_interface_transceiver(
+                            client=self.conn, dId=dev["device_id"], interface=base_port_name
+                        )
+                    port_status = cloudvision.get_interface_status(port_info=port)
+                    port_type = cloudvision.get_port_type(port_info=port, transceiver=transceiver)
+                    if port["interface"] != "":
+                        new_port = self.port(
+                            name=port["interface"],
+                            device=dev["hostname"],
+                            mac_addr=port["mac_addr"],
+                            mode="tagged" if port_mode == "trunk" else "access",
+                            mtu=port["mtu"],
+                            enabled=port["enabled"],
+                            status=port_status,
+                            port_type=port_type,
+                            uuid=None,
+                        )
+                        try:
+                            self.add(new_port)
+                            new_device.add_child(new_port)
+                        except ObjectAlreadyExists as err:
```

Comment:
Do you think we should have a `ValidationError` here too?

Reply:
Sure, we can be on the safe side.

Reply:
Actually, on second thought, that error wouldn't be thrown here. ValidationError is thrown when a `validated_save` is called during the CRUD operations. This is the adapter load so we only need to be concerned with potential duplicate objects being loaded.